core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -667,6 +700,56 @@ public void accept(Path path) {
         return new JarBuildItem(initJar, null, libDir, packageConfig.type, null);
     }
 
+    private void downloadFernflowerJar(PackageConfig packageConfig, Path fernflowerJar) {
+        String downloadURL = String.format("https://jitpack.io/com/github/fesh0r/fernflower/%s/fernflower-%s.jar",
```

Comment:
Is there a reason we download it rather than making it a build dependency?

Reply:
My reasoning is that this is super unlikely to be used by regular users, so there should be no reason to make it an explicit dependency.

Reply:
I see it's not on Maven Central but maybe we could ask nicely?

Reply:
Was going to mention that as well.

For the time being it's only available on Jitpack because some fellow forked the Jetbrains repo and configured Jitpack.

I'll open an issue on the Jetbrains tracker

Reply:
It's a bit more complicated than that. We had quite a lot of corporate users complaining it's hard to download things outside of Maven Central. So downloading from Maven Central is usually the way to go. Not sure downloading it always would be a big issue given how many artifacts people get to download to do a build. But I have no idea how big it is.

Reply:
I opened IDEA-262376.

But let's be clear about the scope of this PR: This is extremely unlikely to be used by anyone not developing Quarkus extensions or Core (more likely only the latter...)

Reply:
Also with Maven, you get signature checking, automatic updates with Dependabot and so on.

It's not on Maven atm so we can't really do anything about it but maybe we could ask?

Reply:
Yup, I did just that in https://youtrack.jetbrains.com/issue/IDEA-262376

Reply:
> But I have no idea how big it is.

For the hash I have used, the jar is around 600kB

Reply:
I was going to suggest making this default to true if the current user is `stephane`, but that jar thing looks like an attack vector :-/

Reply:
Hahahhah